### ISSUE TITLE
Webapps and functions randomly fails "Run From Package" deployments

### ISSUE THREAD

**Issue by tomkuijsten** [projectkudu/kudu#3274]
> For deployment issues, please provide us with the following information:
> ---------------------------------------------------------------

### Repro steps.
We use Azure Devops to deploy to Azure Webapps and Functions. For functions, we use the AzureFunctionApp@1 task set to 'auto' as deployment method. In the logs I can see it tries to do a "Run From Package" deployment, which is exactly what I want. In Azure Devops the step is marked as succeeded, however, when I look at the "Deployment Center" of the function app, the deployment is marked as "Failed" en the old code is still running.

There are 2 problems here:
1. The task is marked as success, which is not true. I already added an [issue](https://github.com/microsoft/azure-pipelines-tasks/issues/14495) for the azure-pipeline repo and they admitted that it's an issue on their side and they will fix that.
2. The deployment fails, but I don't understand why. If I dig into the kudu logs, I can find the following:

[{"log_time":"2021-03-01T19:56:09.4193388Z","id":"5b1c1401-a3dc-41ef-85ef-2abb46b89dfa","message":"An unknown error has occurred. Check the diagnostic log for details.","type":2,"details_url":null}]

And when digging deeper in the kudu logs I find this error:

<step title="Error occurred" date="2021-03-01T19:56:02.115" type="error" text="Thread was being aborted." stackTrace="   at Kudu.Core.Infrastructure.OperationManager.Attempt[T](Func`1 action, Int32 retries, Int32 delayBeforeRetry, Func`2 shouldRetry) in C:\Kudu Files\Private\src\master\Kudu.Core\Infrastructure\OperationManager.cs:line 47 (and a whole bunch of extra exception lines)

### The log/error given by the failure.

Here are all the relevant log files I could find:
[DeployProblemsAzure.zip](https://github.com/projectkudu/kudu/files/6075005/DeployProblemsAzure.zip)

### Debug your Azure website remotely.
Our function name is: dLonWeuConMonFa01

### Mention any other details that might be useful.
It seems to be very random behavior. If we deploy the function a second time, most times it succeeds.

**Comment by suwatch** [comment]
From the link (https://github.com/microsoft/azure-pipelines-tasks/issues/14495) - it seems like a devops issue.

**Comment by tomkuijsten** [comment]
I dont think so, devops is reporting the state incorrect, that's also an issue and reported in the mentioned issue. But the deployment did actually fail, as reported in the Azure portal. So I am confused by your comment, the devops task only triggers a kudu task to deploy, right? Devops itself only asks for the state of the deployment to mark the task as finished/success/fail.

**Comment by suwatch** [comment]
Your issue was caused by an update in WEBSITE_RUN_FROM_PACKAGE on and off.   This setting drastically impacts how the file system mounted as well as deployment behaviors.   It is not intended and there is no scenario that one would turn that on/off frequently thru out the lifetime of the app.   This situation happens to your app inadvertently.

During deployment, there is an update appsettings (part of your deployment script) which lacks of such setting (meaning to turn off WEBSITE_RUN_FROM_PACKAGE).   After which, since the RunFromPackage option is checked, the DevOps is trying to turn in back on before deploying.   This caused the confusion in deployment and, often times, fails the deployment altogether.   

If your intention is to use WEBSITE_RUN_FROM_PACKAGE, make sure to include WEBSITE_RUN_FROM_PACKAGE=1 in your update appSettings as well (to avoid flip-flop).   We are working with DevOps to provide better experience to this.

**Comment by tomkuijsten** [comment]
@suwatch now that's an answer! Thanks a lot, just fixed the problem and everything seems to work fine now!
